File: 设计模式/程序/23-TemplatePattern/TemplatePattern(python)/FootballGame.py
from Game import Game 

class FootballGame(Game):
    # 实现三个函数
    def Initial(self):
        print('Prepare Football Ground...')
        print('Wait Players...')
        return True

    def PlayGame(self):
        print('Playing...')
        print('tititititi')
        return 3

    def EndGame(self):
        print('Player Take Shower...')
        print('Clean Football ground...')

File: 设计模式/程序/23-TemplatePattern/TemplatePattern(C++)/src/FootballGame.cpp
#include "FootballGame.h"
#include <iostream>

namespace TemplatePatternApp
{
        bool FootballGame::Initialize() const 
        {
            std::cout << "Wait Players...\n";
            std::cout << "Prepare Football Ground...\n";
            return true;
        }
        int FootballGame::PlayGame() const 
        {
            std::cout << "Playing...\n";
            std::cout << "titititti\n";
            return 5;
        }
        void FootballGame::EndGame() const 
        {
            std::cout << "Players Take Shower...\n";
            std::cout << "Clean Football Ground...\n";
        }
}